heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/MetricsCacheManagerHttpServer.java

```diff
@@ -0,0 +1,148 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.logging.Logger;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import com.twitter.heron.metricscachemgr.metricscache.MetricsCache;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.utils.NetworkUtils;
+
+
+/**
+ * MetricsCacheMgr http server:
+ * compatible with tmaster and tracker http interface for metrics
```

Comment:
done

  Differece from MetricsCacheManagerServer:
  1. MetricsCacheManagerServer accepts metric publishing message from sinks;
   MetricsCacheManagerHttpServer responds to queries.
  2. MetricsCacheManagerServer is a HeronServer;
   MetricsCacheManagerHttpServer is a http server